File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> mp = new HashMap<>();
        int[] ans = new int[2];
        for (int i = 0; i < nums.length; i++) {
            mp.put(nums[i],i);
        }

        for (int i = 0; i < nums.length; i++) {
            if(mp.containsKey(target - nums[i]) && i != mp.get(target - nums[i]))
            {
                ans[0] = i;
                ans[1] = mp.get(target - nums[i]);
                break;
            }
        }
        return ans;
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp:Dict[int,int] = {}
        ans:List[int] = []

        for i in range (0,len(nums)):
            mp[nums[i]] = i

        for i in range (0,len(nums)):
            if((target - nums[i] in mp) and i != mp[target - nums[i]]):
                ans.append(i)
                ans.append(mp[target - nums[i]])
                break
        return ans